formats/properties/commonMain/src/kotlinx/serialization/properties/StringProperties.kt

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright 2017-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.properties
+
+import kotlinx.serialization.*
+import kotlinx.serialization.descriptors.*
+import kotlinx.serialization.encoding.*
+import kotlinx.serialization.internal.*
+import kotlinx.serialization.modules.*
+
+/**
+ * Transforms a [Serializable] class' properties into a single flat [String] representing the class data
+ * in the properties format.
+ *
+ * If the given class has non-primitive property `d` of arbitrary type `D`, `D` values are inserted
+ * into the same map; keys for such values are prefixed with string `d.`:
+ *
+ * ```
+ * @Serializable
+ * class Data(val property1: String)
+ *
+ * @Serializable
+ * class DataHolder(val data: Data, val property2: String)
+ *
+ * val string = StringProperties.encodeToString(properties)
+ * // string contents will be the following:
+ * """
+ *     property2 = value2
+ *     data.property1 = value1
+ * """
+ * ```
+ *
+ * If the given class has a [List] property `l`, each value from the list
+ * would be prefixed with `l.N.`, where N is an index for a particular value.
+ * [Map] is treated as a `[key,value,...]` list.
+
+ * Conversely, this class can convert a properties string into a [Serializable] class instance.
+ * ```
+ * @Serializable
+ * class Data(val property1: String)
+ *
+ * @Serializable
+ * class DataHolder(val data: Data, val property2: String)
+ *
+ * val string = """
+ *     property2 = value2
+ *     data.property1 = value1
+ * """
+ * val data = StringProperties.decodeToString(string, DataHolder.serializer())
+ * // data contents will be the following:
+ * // DataHolder(data = Data(property1 = "value1"), property2 = "value2")
+ * ```
+ *
+ * @param conf A [PropertiesConf] which contain configuration for customising the output string.
+ */
+@ExperimentalSerializationApi
+@Suppress("UNUSED_PARAMETER")
+public sealed class StringProperties(
+    private val conf: PropertiesConf,
+    private val properties: Properties = Properties(conf.serializersModule),
+) : SerialFormat by properties, StringFormat {
+
+    /**
+     * Encodes properties from the given [value] to a properties String using the given [serializer].
+     * `null` values are omitted from the output.
+     */
+    @ExperimentalSerializationApi
+    public override fun <T> encodeToString(serializer: SerializationStrategy<T>, value: T): String {
+        val map = properties.encodeToMap(serializer, value)
+        val builder = StringBuilder()
```

Comment:
I would extract the actual output renderer into another function taking a map overload to encode a giving String map to a Properties file
```@ExperimentalSerializationApi
    public fun <T> encodeToString(map: Map<String, String>): String {
```